tensorboard/webapp/widgets/line_chart_v2/line_chart_component.ts

```diff
@@ -182,8 +182,16 @@ export class LineChartComponent implements AfterViewInit, OnChanges, OnDestroy {
   }
 
   ngAfterViewInit() {
+    const dimPropChanged = this.readAndUpdateDomDimensions();
     this.initializeChart();
-    this.updateLineChart();
+    const viewboxPropChanged = this.updateLineChart();
+
+    // After view is initialized, if we ever change the Angular prop that should propagate
+    // to children, we need to retrigger the Angular change. Since we lazily update the
+    // property, these may return false.
+    if (dimPropChanged || viewboxPropChanged) {
+      this.changeDetector.detectChanges();
+    }
```

Comment:
iiuc, the idea is that when Angular calls our `ngAfterViewInit` lifecycle hook, it assumes we won't alter any props used in the template.  Since we do alter props used in the template, we need the explicit `detectChanges()` call.

In this PR, we are careful to only call detectChanges when necessary, but I think this adds a bit much complexity.
- L186 seem to be order-dependent, since `initializeChart` internally calls `readAndUpdateDomDimensions`, so it must come before L185.
- `updateLineChart` returns `ngStateUpdated`, but it only depends on the `viewBox` not any other state changes
- `ngStateUpdated` is not true when the viewBox changes to the 'fixed' or 'overridden' value.  Are we sure this is safe?

I think it would be a lot simpler to just call `detectChanges()` here in `ngAfterView` unconditionally, without checking whether the dim/viewbox actually changed.  If we ever updated the code to remove the `viewBox` from the template, or added new props to the template that can be updated from `ngAfterViewInit`, it would be less fragile to those changes as well.

wdyt?

Reply:
Reverted most changes and only kept the unconditional `detectChanges`.